Add `ArraySlice` & `StringSlice` type #734 (#748)
sindresorhus/type-fest: readme.md

- [`Trim`](source/trim.d.ts) - Remove leading and trailing spaces from a string.
- [`Split`](source/split.d.ts) - Represents an array of strings split using a given character or character set.
- [`Replace`](source/replace.d.ts) - Represents a string with some or all matches replaced by a replacement.

> - [`StringSlice`](source/string.d.ts) - Returns a string slice of a given range(with negative index support), just like `String.slice` in js.

Reviewer: `in js` is moot

Author: removed

> - [`StringSlice`](source/string.d.ts) - Returns a string slice of a given range(with negative index support), just like `String.slice` in js.

Reviewer: ```suggestion
- [`StringSlice`](source/string.d.ts) - Returns a string slice of a given range(with negative index support), just like `String#slice` in js.
```

Reviewer: > with negative index support

This is not important enough to have here. can be in the file doc comment instead.

Author: updated

- [`NonNegative`](source/numeric.d.ts) - A non-negative `number`/`bigint` (`0 <= x < ∞`).
- [`NegativeInteger`](source/numeric.d.ts) - A negative (`-∞ < x < 0`) `number` that is an integer.
- [`NonNegativeInteger`](source/numeric.d.ts) - A non-negative (`0 <= x < ∞`) `number` that is an integer.
- [`IsNegative`](source/numeric.d.ts) - Returns a boolean for whether the given number `T` is a negative number.

> - [`Add`](source/numeric.d.ts) - Return the result of `A + B`.

Reviewer: Needs a better description.

Author: updated

- [`NonNegative`](source/numeric.d.ts) - A non-negative `number`/`bigint` (`0 <= x < ∞`).
- [`NegativeInteger`](source/numeric.d.ts) - A negative (`-∞ < x < 0`) `number` that is an integer.
- [`NonNegativeInteger`](source/numeric.d.ts) - A non-negative (`0 <= x < ∞`) `number` that is an integer.
- [`IsNegative`](source/numeric.d.ts) - Returns a boolean for whether the given number is a negative number.

> - [`GreaterThan`](source/greater-than.d.ts) - Returns a boolean for whether A is greater than B.

Reviewer: ```suggestion
- [`GreaterThan`](source/greater-than.d.ts) - Returns a boolean for whether a given number is greater than another number.
```

Same with the others.

Author: done

- [`GreaterThanOrEqual`](source/greater-than-or-equal.d.ts) - Returns a boolean for whether A is greater than or equal to B.
- [`LessThan`](source/less-than.d.ts) - Returns a boolean for whether A is less than B.
- [`LessThanOrEqual`](source/less-than-or-equal.d.ts) - Returns a boolean for whether A is less than or equal to B.

> - [`Add`](source/add.d.ts) - Returns the sum of the two input numbers.

Reviewer: ```suggestion
- [`Add`](source/add.d.ts) - Returns the sum of two numbers.
```

> - [`Subtract`](source/subtract.d.ts) - Returns the difference between two input numbers.

Reviewer: ```suggestion
- [`Subtract`](source/subtract.d.ts) - Returns the difference between two numbers.
```

Author: done
